core/VectorUopGenerator.cpp

```diff
@@ -76,6 +76,14 @@ namespace olympia
         // Set weak pointer to parent vector instruction (first uop)
         sparta::SpartaWeakPointer<olympia::Inst> weak_ptr_inst = current_inst_;
         uop->setUOpParent(weak_ptr_inst);
+        uop->setEEW(current_inst_->getEEW());
+        uop->setMOP(current_inst_->getMOP());
+        uop->setStride(current_inst_->getStride());
+        if(uop->isLoadStoreInst()){
+            // set base address according to LMUL, i.e if we're on the 3rd
+            // LMUL Uop, it's base address should be base address + 3 * EEW
+            uop->setTargetVAddr(uop->getTargetVAddr() + uop->getEEW() * uop->getUOpID());
```

Comment:
This is fine for now since this PR only supports unit stride, but eventually we should create an "address unroller" class that generates all of the addresses for a vector uop. I think it makes sense to be a part of this class, but it could also be a part of the VLSU.